### DISCUSSION TITLE
Having trouble in installing with Homebrew on macOS M2 pro

### DISCUSSION THREAD

**Discussion by maryam-khd** [firedrakeproject/firedrake#3492] [Firedrake support]
Hello, 
I have followed every steps for installation on macOS via Homebrew and everything seems fine until I check the test, as suggested,
cd $VIRTUAL_ENV/src/firedrake
pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection"
then I encounter error for MPI version
adding the 
127.0.0.1       LOCALHOSTNAME.local
127.0.0.1       LOCALHOSTNAME
also did not help. I am not sure that is there any step I have to take consequently. 

My system had the anaconda during installation but I also deleted the anaconda related from  ~/.zshrc file and by reseting the system still the problem resist. 
I am wondering how I can start over by correctly delete/uninstall firedrake package. 
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/14887235/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/14887255/configure.log)
[make.log](https://github.com/firedrakeproject/firedrake/files/14887260/make.log)

**Comment by Ig-dolci** [comment]
Your `firedrake-install.log` shows:
```
2024-04-05 11:43:34,763 WARNING Specified venv '/Users/maryam/firedrake' already exists
2024-04-05 11:43:34,763 ERROR  Can't install into existing venv '/Users/maryam/firedrake'
2024-04-05 11:43:34,764 INFO 
```
So, you need first delete the repository `/Users/maryam/firedrake` and try to install firedrake again.

**Comment by maryam-khd** [comment]
Thanks for your response. 
Is deleting enough for uninstalling?

**Reply by connorjward** [reply]
You must also deactivate the previous virtual environment if it was activated (run `deactivate`). But otherwise that's all you need to do.

**Comment by maryam-khd** [comment]
Thanks for you suggestions, but deleting the Anaconda from $PATH and reinstalling the firedrake I still encounter following error by running the test, 
[terminaloutput.txt](https://github.com/firedrakeproject/firedrake/files/14923136/terminaloutput.txt)
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/14923141/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/14923161/configure.log)
[make.log](https://github.com/firedrakeproject/firedrake/files/14923163/make.log)

**Reply by connorjward** [reply]
This is bizarre. You are only failing the parallel tests. I do notice though that you are not using a homebrew installed Python when you run `firedrake-install`. The MacOS default Python is known to cause issues so maybe running `$(brew --prefix)/bin/python3 firedrake-install` might help you.
